### PR TITLE
NS4-22 Saving weights complete need review

### PR DESCRIPTION
**Pull request by LouisOporto**
Review the saving weights to the project. Two methods by best val_loss and just precautionary saving per round

### CODE REVIEW

**Comment by LouisOporto** [comment]
Looks good. The best weights based on validation loss is saved between epochs. Sessions have persistent best weights between rounds.
